### ISSUE TITLE
scan works different if executed from script and from command line 

### ISSUE THREAD

**Issue by vgrebenschikov** [jqlang/jq#3217]
**Describe the bug**
scan() returns unexpected results when executed from shbang file:

test.jq file:
```shell
$ cat test.jq
#!/usr/bin/jq -Rn -f
inputs | scan("(\"(?:[^\"]*(?:\"\"[^\"]*)*)\"|(?:[^,]+)|(?<=,)(?=,))")
```
 
execution:
```shell
$ echo 'first,,,next' | ./test.jq
[
  "first"
]
""
""
""
[
  "next"
]
$
```

Notice, three empy strings in middle - which does not matches regexp provided

while in pure command line run of jq the same scan works as expected 
- all output of scan() is arrays (unlike just strings above), and only two elements matching comma-comma pairs:

```shell 
$ echo 'first,,,next' | jq -nR 'inputs | scan("(\"(?:[^\"]*(?:\"\"[^\"]*)*)\"|(?:[^,]+)|(?<=,)(?=,))")'
[
  "first"
]
[
  ""
]
[
  ""
]
[
  "next"
]
```

**To Reproduce**
See above reproduction 

**Expected behavior**
in both cases following returned:
```json
[
  "first"
]
[
  ""
]
[
  ""
]
[
  "next"
]
```

**Environment (please complete the following information):**

- macOS 15.1.1
- jq-1.7.1

**Additional context**

**Comment by pkoppstein** [comment]
I was able to reproduce the described behavior in a mac environment, but the problem seems to have been resolved as of jq-1.7.1-79-g96e8d89

**Comment by wader** [comment]
Same for me, can't reproduce with 1.7.1 but can with 1.6

@vgrebenschikov what does `/usr/bin/jq --version` say?

**Comment by vgrebenschikov** [comment]
> Same for me, can't reproduce with 1.7.1 but can with 1.6
> 
> @vgrebenschikov what does `/usr/bin/jq --version` say?

yes, shbang refer to macOS included jq - which is 1.6 - my bad

```shell
$ /usr/bin/jq --version
jq-1.6-159-apple-gcff5336-dirty
```

so my simpliied example not triggered any more on 1.7.1 but somehow original problem (not simplified) - still triggered - will return with more details

**Comment by vgrebenschikov** [comment]
ok, let us assume that problem was fixed, 
other problem related to BOM character at begining of CSV that I can't cut of with regexp somehow 

fix about BOM character is here #2137
I've w/a it with `sub("^\uFEFF"; "")`

**Comment by wader** [comment]
> > Same for me, can't reproduce with 1.7.1 but can with 1.6
> > @vgrebenschikov what does `/usr/bin/jq --version` say?
> 
> yes, shbang refer to macOS included jq - which is 1.6 - my bad

No worries 👍 i'm guessing it got fix by https://github.com/jqlang/jq/pull/2724

> so my simpliied example not triggered any more on 1.7.1 but somehow original problem (not simplified) - still triggered - will return with more details

That would be great
